server/src/localcache.rs

```diff
@@ -0,0 +1,189 @@
+/*
+ * Parseable Server (C) 2022 - 2023 Parseable, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+use std::{io, path::PathBuf, sync::Arc};
+
+use fs_extra::file::CopyOptions;
+use futures_util::TryFutureExt;
+use hashlru::Cache;
+use itertools::{Either, Itertools};
+use object_store::ObjectStore;
+use once_cell::sync::OnceCell;
+use tokio::{fs, sync::Mutex};
+
+use crate::option::CONFIG;
+
+pub const STREAM_CACHE_FILENAME: &str = ".cache.json";
+
+#[derive(Debug, serde::Deserialize, serde::Serialize)]
+pub struct LocalCache {
+    version: String,
+    current_size: u64,
+    capacity: u64,
+    files: Cache<String, PathBuf>,
+}
+
+impl LocalCache {
+    fn new_with_size(capacity: u64) -> Self {
+        Self {
+            version: "v1".to_string(),
+            current_size: 0,
+            capacity,
+            files: Cache::new(100),
```

Comment:
Why 100 only here?

Reply:
backing hashlru is capacity based. The capacity is arbitrary and is increased when needed.

Reply:
who increases? user or the backing cache?